### PR TITLE
fix(git): `.github.io` in package URL is incorrectly replaced as `hub.io`

### PR DESCRIPTION
**Pull request by ikeyan**
**Summary**
When I executed yarn with a package.json like below, I got a following error.
```
error Command failed.
Exit code: 128
Command: git
Arguments: ls-remote --tags --heads ssh://git@github.com/kawashimaken/ajaxzip3hub.io.git
Directory: /Users/me/foo
Output:
ERROR: Repository not found.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```
The point is, yarn tried to access "ajaxzip3hub.io.git", but "ajaxzip3.github.io.git" was expected.
This PR fixes this issue.

package.json:
```JSON
{
  (omitted)
  "dependencies": {
    "@bower_components/ajaxzip3": "kawashimaken/ajaxzip3.github.io#^1.0.0"
  }
}
```
( https://github.com/kawashimaken/ajaxzip3.github.io exists)

**Test plan**

Added a new test case. Also existing test cover most of the functionality.

### CODE REVIEW

File: src/resolvers/exotics/hosted-git-resolver.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ export function explodeHostedGitFragment(fragment: string, reporter: Reporter):
     if (repoParts.length <= 3) {
       return {
         user,
-        repo: repoParts[0].replace('.git', ''),
+        repo: repoParts[0].replace(/[.]git$/, ''),
```
Comment (BYK) [inline src/resolvers/exotics/hosted-git-resolver.js]: Use `\.` instead of `[.]` please :)

Comment (ikeyan) [reply src/resolvers/exotics/hosted-git-resolver.js]: Fixed.

**Comment by ikeyan** [comment]
I added two test cases, one for the title of this PR and the other for a package URL ending with '.git.git'

File: src/resolvers/exotics/hosted-git-resolver.js (hunk 1 above)
Comment (ikeyan) [reply src/resolvers/exotics/hosted-git-resolver.js]: @BYK Fixed.

**Comment by ikeyan** [comment]
@BYK Please review again... I can't migrate from bower until this PR is merged...

**Comment by iansu** [comment]
I believe I fixed this in #5119 which has been merged.

**Comment by ikeyan** [comment]
Agreed

**Pull request closed without merging**